bin/rose-check-software

```diff
@@ -208,43 +247,92 @@ def check_all(name, dep_list):
         bool - True if all checks passed, False otherwise.
     """
     print name
-    print '-' * len(name)
-    for msg, _ in dep_list:
-        print msg
-    if all(result for _, result in dep_list):
-        print 'Result: PASS\n'
-        return True
+    if len(dep_list[0]) == 2:
+        print '-' * len(name)
+        # Performing dependency checking.
+        for msg, _ in dep_list:
+            print msg
+        if all(result for _, result in dep_list):
+            print 'Result: PASS\n'
+            return True
+        else:
+            print 'Result: FAIL\n'
+            return False
     else:
-        print 'Result: FAIL\n'
-        return False
+        # Outputting dependency list for documentation.
+        for msg in dep_list:
+            print '   %s' % msg
+        print
 
 
-def main():
+def main(check=check):
     """Check required and optional dependencies."""
     ret = check_all('Required Software', [
+        check('python', (2, 6, 6), (3,), version_template=r'Python (.*)',
+              outfile=2),
+        check('cylc'),
+        check('py:pygtk<gtk>', (2, 12), attr_name='pygtk_version'),
+        check('py:cherrypy'),
+        check('py:jinja2'),
+        check('py:requests', (2, 2, 1)),
+        check('py:sqlalchemy', (1, 1)),
+    ])
+
+    check_all('Rosie Dependencies', [
+        check('svn', (1, 8), command_template=['--version', '--quiet']),
+        check('fcm', version_template='FCM ([\d\.\-]+)'),
+        check('perl', (5, 10, 1), command_template='-v',
+              version_template=r'This is perl, v([^\s]+)')
+    ])
+
+    docs(check=check)
+    return ret
+
+
+def docs(check=check):
+    """Check software dependencies for the documentation builder."""
+    ret = check_all('Documentation Builder', [
         check('python', (2, 7), (3,), version_template=r'Python (.*)',
                outfile=2),
         check('dot', command_template='-V',
```

Comment:
... in fact since this gets loaded into the 'installation' page of the docs this is related strictly to this PR; 'dot' is pretty meaningless when listed under 'system requirements'.

Reply:
In fact, _ideally_ we can do this so all software requirements have their names as loaded into the docs to include given capitalisation e.g. 'TeX', 'CherryPy' etc instead of in all-lowercase - so take a pair of arguments, one text label and one required command, perhaps? Just to keep it all looking professional :+1:

Reply:
I have listed the command names - hence lower case. Note that there is no graphviz command and the dependency is on `dot` as there are different graphviz layout engines.

Reply:
Fair enough. RE graphviz, I know that - I am trying to say 'graphviz' should be reported to the user as the module name, not 'dot'.

Reply:
Ok, have gone with `graphviz-dot <dot>`.